samza-api/src/main/java/org/apache/samza/table/descriptors/RemoteTableDescriptor.java

```diff
@@ -305,6 +307,14 @@ public String getProviderFactoryClassName() {
       }
     }
 
+    //emit table api read/write rate limit
+    if (this.enableReadRateLimiter && tagCreditsMap.containsKey(RL_READ_TAG)) {
```

Comment:
s/emit/write.

Can you extract all of the rate limiter related config generation logic (factory, serde, limits) to a helper method? Will be easier to read.